### ISSUE TITLE
Querying array values

### ISSUE THREAD

**Issue by haasted** [tidwall/gjson#93]
Thanks for a great framework!

Can the query syntax be applied to arrays, or is it solely applicable to objects?

My scenario resembles the following:
```
{
    "artists": [
        "Bob Dylan",
        "John Lennon",
        "Mick Jagger",
        "Elton John",
        "Michael Jackson",
        "John Smith"
    ]
}
```

Is there any way to use the query syntax to exctract values from this list? Something along the lines of:

```
artists.#[value % "John"] >> "John Lennon"
artists.#[value % "John"]# >> ["John Lennon", "Elton John", "John Smith"]
```

**Comment by tidwall** [comment]
Right now queries are only for objects. But I think it makes sense that it could be modifies to work on arrays too.

**Comment by tidwall** [comment]
This is now available in v1.2.2

A standard query without a name is all that's needed.

```
artists.#[%"John*"]     >> "John Lennon"
artists.#[%"John*"]#   >> ["John Lennon","John Smith"]
```
